### DISCUSSION TITLE
How to add GraphQL operations dynamically to the schema

### DISCUSSION THREAD

**Discussion by asegurat** [strawberry-graphql/strawberry#3555] [Q&A]
How can I dynamically add GraphQL operations (_mutations_ and _queries_) to a schema composed of static operations?

That is, once I have created my schema and added the `Query` and `Mutation` classes to it, how can I expand it, adding operations dynamically?

Specifically in my case, I am using Federation 2.

```python
@strawberry.type
class Mutation:

    @strawberry.field
    def my_namespace() -> MyMutation:
        return InjectorService.get_instance(MyMutation)


@strawberry.type
class Query:

    @strawberry.field
    def my_namespace() -> MyQuery:
        return InjectorService.get_instance(MyQuery)

schema = gf.Schema(query=Query, mutation=Mutation, enable_federation_2=True)
graphql_api = GraphQLRouter(schema, graphql_ide=True)
```

Thanks in advance.

cc: @patrick91

**Comment by YuriiMotov** [comment]
You can add types to the schema dynamically, but only before your server started.
To do that you can use `create_type`: https://strawberry.rocks/docs/guides/tools

As I understand it, you can not modify the schema after starting the server.
